### PR TITLE
Add JMH test sources to build to detect compilation errors early

### LINKED ISSUE #1141: Compile JMH sources in install phase on Jenkins
Currently a compilation error in the JMH sources will only be detected during the JMH phase will the test suite is running in parallel. Adding the `jmh` profile to the install phase does not work as it would also execute the JMH tests immediately. Find a way to add the JMH sources to the compilation without executing it.

### PR DESCRIPTION
**Pull request by menski**
closes #1141

### CODE REVIEW

**Comment by menski** [comment]
@Zelldon this will automatically add the JMH sources as test sources folder in Intellij. Why was this a problem for you? Sorry I always forget the reason. :see_no_evil:

**Comment by Zelldon** [comment]
My problem was that I was not able to compile the  jmh test, because they have not being maintained.
If I wanted to run simply the tests, after a `mvn clean install`, Intellij reimported the jmh modules again and I had to exclude them before executing the tests.

But if we now maintain them and compile them on every build I shouldn't have any problems with that.

**Review by ChrisKujawa — APPROVED** [review]

File: parent/pom.xml [hunk 1]
```diff
@@ -474,6 +496,14 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>jmh-compile</id>
```
[the remark this answers is not in the document]
Comment (menski) [reply parent/pom.xml]: yeah I forgot to delete my initial approach :woman_facepalming:

**Pull request merged**